tests/cli-misc.rs

```diff
@@ -785,13 +786,26 @@ fn completion_cargo_unsupported_shell() {
 }
 
 #[test]
-fn set_profile() {
+fn set_profile_doesnt_change_installed_components() {
     setup(&|config| {
         expect_stdout_ok(config, &["rustup", "show", "profile"], "default");
-        assert!(config.exedir.join(format!("rls{}", EXE_SUFFIX)).exists());
+        expect_ok(
+            config,
+            &[
+                "rustup",
+                "toolchain",
+                "install",
+                "--no-self-update",
+                "stable",
+            ],
```

Comment:
Why did you add this `toolchain install`?

Reply:
Because I wanted to demonstrate that with a toolchain installed, changing the profile didn't cause components to auto-install when executed.  Does that make sense?